### ISSUE TITLE
false `use_super_parameters` lint

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#3569]
The code blow produces a `use_super_parameters` lint. AFAIK, there is no way to express this with super parameters. Therefore, no lint warning should be produced.

```dart
class Rect {
  Rect(this.width, this.height);

  final double width;
  final double height;
}

class Square extends Rect {
  // 🔥 info: Convert 'dimension' and 'dimension' to super parameters. (use_super_parameters at [playground] lib/main.dart:9)
  Square(double dimension) : super(dimension, dimension);
}
```

**Comment by pq** [comment]
Thanks for this repro!
